Miscellaneous Cleanups

dopplershift opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/Unidata/siphon/blob/main/CONTRIBUTING.md
-->

#### Description Of Changes
* Remove outdated Travis (and AppVeyor) info
* Remove outdated `TODO.md`
* Make linkchecker pass (Fixes #828)
* Clean up deprecation warnings from `utcnow()`
* Clean up suppression of warnings related to setting `.units` on `DataFrame`s
* Make test suite run with warnings as errors

<!--
Feel free to remove check-list items aren't relevant to your change

Please use keywords (e.g., Fixes, Closes) to create link to the issues or pull
requests you resolved, so that they will automatically be closed when your pull
request is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

#### Checklist

- [x] Closes #828
- [x] Finishes last item from #270 (cleaning up warnings)
- [x] Fully documented

dcamron reviewed (changes requested):
I guess they don't call it codegrammar

dopplershift commented on docs/developerguide.rst:
```diff
@@ -116,12 +116,19 @@ checked with ``doc8``. To see what the docs will look like, you also need to ins
+and Python versions, as well as checks for any broken web links. ``flake8`` (along with a
+variety of plugins found in ``ci/linting.txt``) and ``ruff``are also run against the code to
+check formatting using another job on GitHub Actions. As part of this linting job, the docs
+are also checked using the ``doc8`` tool, and spelling is checked using the ``codespell``.
```
Copy/pasted from MetPy 😆

dopplershift commented:
In addition to addressing the review comments, I went ahead and pushed a commit that enables Sphinx nitpicky mode and cleans up all of the issues that identified.
